apps/server/src/utils/institutionResolver.test.ts

```diff
@@ -1,120 +1,311 @@
+import * as preferences from '../shared/preferences'
 import { elasticSearchInstitutionData } from '../test/testData/institution'
 import { ElasticSearchMock } from '../utils/ElasticSearchClient'
-import { getAvailableProviders, resolveInstitutionProvider } from './institutionResolver'
-
-describe('getAvailableProviders', () => {
-  it('gets mx and sophtron providers', () => {
-    const expectedProviders = ['mx', 'sophtron']
-    const institution = {
-      ...elasticSearchInstitutionData,
-      mx: {
-        ...elasticSearchInstitutionData.mx,
-        id: 'mx_bank'
-      },
-      sophtron: {
-        ...elasticSearchInstitutionData.sophtron,
-        id: 'sophtron_bank'
+import {
+  getAvailableProviders,
+  resolveInstitutionProvider
+} from './institutionResolver'
+import testPreferences from '../../cachedDefaults/testData/testPreferences.json'
+
+const mockInstitutionWithMxAndSophtron = (institutionId = 'test') => {
+  ElasticSearchMock.add(
+    {
+      method: 'GET',
+      path: `/institutions/_doc/${institutionId}`
+    },
+    () => {
+      return {
+        _source: {
+          ...elasticSearchInstitutionData,
+          is_test_bank: false,
+          mx: {
+            id: 'mx_id'
+          },
+          sophtron: {
+            id: 'sophtron_bank'
+          }
+        }
       }
     }
+  )
+}
 
-    expect(getAvailableProviders(institution)).toEqual(expectedProviders)
-  })
-
-  // This is temporary until we fully support finicity and akoya
-  it('only returns mx and sophtron even if finbank and akoya are configured', () => {
-    const institution = {
-      ...elasticSearchInstitutionData,
-      mx: {
-        ...elasticSearchInstitutionData.mx,
-        id: 'mx_bank'
-      },
-      sophtron: {
-        ...elasticSearchInstitutionData.sophtron,
-        id: 'sophtron_bank'
-      },
-      finicity: {
-        ...elasticSearchInstitutionData.finicity,
-        id: 'finicity_bank'
-      },
-      akoya: {
-        ...elasticSearchInstitutionData.akoya,
-        id: 'akoya_bank'
+const mockInstitutionWithMx = (institutionId = 'test') => {
+  ElasticSearchMock.add(
+    {
+      method: 'GET',
+      path: `/institutions/_doc/${institutionId}`
+    },
+    () => {
+      return {
+        _source: {
+          ...elasticSearchInstitutionData,
+          is_test_bank: false,
+          mx: {
+            id: 'mx_id'
+          },
+          sophtron: {
+            id: null
+          }
+        }
       }
     }
-    const expectedProviders = ['mx', 'sophtron']
+  )
+}
 
-    expect(getAvailableProviders(institution)).toEqual(expectedProviders)
+describe('institutionResolver', () => {
+  beforeEach(() => {
+    jest
+      .spyOn(preferences, 'getPreferences')
+      .mockResolvedValue(testPreferences as preferences.Preferences)
```

Comment:
This is a concept I'm familiar with through ruby tests but didn't know how to do in jest or if it was even possible, so it's nice to see it.